src/utils/storage-helper.js

```diff
@@ -1,13 +1,21 @@
 import { IMAGE_DOWN_SCALING_WIDTH } from '../domain/constants.js';
 
-const getScaledDimensions = (img, width) => {
+const getScaledDownDimensions = (img, width) => {
+  if (img.naturalWidth <= width) {
+    return { width: img.naturalWidth, height: img.naturalHeight };
+  }
   const ratio = img.naturalWidth / width;
   return { width, height: Math.round(((img.naturalHeight / ratio) + Number.EPSILON) * 100) / 100 };
 };
 
 const scalableFileTypes = ['image/png', 'image/jpg', 'image/jpeg', 'image/webp'];
 
-const scaleDownImage = ({ file, width }) => {
+const shouldOptimizeImage = ({ naturalSize, naturalWidth, width }) => {
+  const optimize = (naturalWidth > width && (naturalSize > 500 * 1000)) || (naturalSize > 1.5 * 1000 * 1000);
```

Comment:
It's maybe a little bit inconsistent that the `width` threshold is parameterized, but the `size` (in bytes) thresholds are hard-coded. We could put them also into `constants.js`.

Reply:
That's true, I extracted all 3 limits and named them clearly, then used them directly in the `storage-helper`. Thanks for pointing this out.

Reply:
Cool, thanks :+1: